service/history/api/recordactivitytaskstarted/api.go

```
@@ -281,3 +306,50 @@ func getDeploymentVersionForWorkflowId(
 	current, ramping := worker_versioning.CalculateTaskQueueVersioningInfo(tqData.GetDeploymentData())
 	return worker_versioning.FindDeploymentVersionForWorkflowID(current, ramping, workflowId), nil
 }
+
+func processActivityWorkflowRules(
+	shardContext historyi.ShardContext,
+	request *historyservice.RecordActivityTaskStartedRequest,
+	ms historyi.MutableState,
+	ai *persistencespb.ActivityInfo,
+) (rejectCode, error) {
+	if ai.Stamp == request.Stamp && ai.Paused {
+		// this shouldn't happen. For now log an error
```

I thought you want to return an error.